source/payload-interfaces/index.rst

```diff
@@ -86,167 +109,207 @@ type EFI_RESOURCE_MEMORY_RESERVED.
 I/O and memory mapped I/O resource should also be reported using
 resource type EFI_RESOURCE_IO and EFI_RESOURCE_MEMORY_MAPPED_IO.
 
-**Open**: should report payload in memory using the Boot Firmware
-Volume (BFV) HOB?
+Memory Allocation HOB
+^^^^^^^^^^^^^^^^^^^^^
+
+The bootloader should report the memory usages that exist outside the
+HOB list through the HOB following EFI_HOB_MEMORY_ALLOCATION format defined
+in *Platform Initialization Specification Volume 3: Shared Architectural
+elements*.
+
+Boot-Strap Processor (BSP) Stack Memory Allocation HOB
+''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+The bootloader should report the initial stack prepared for payload through
+the HOB following EFI_HOB_MEMORY_ALLOCATION_STACK format defined in *Platform
+Initialization Specification Volume 3: Shared Architectural elements*.
+
+Memory Allocation Module HOB
+''''''''''''''''''''''''''''
+
+The bootloader should report the payload memory location and entry point
+through the HOB following EFI_HOB_MEMORY_ALLOCATION_MODULE format defined
+in *Platform Initialization Specification Volume 3: Shared Architectural
+elements*.
+
+Graphics information HOB
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If bootloader initializes the graphics device, the bootloader might
+report graphics mode and framebuffer information through
+EFI_PEI_GRAPHICS_INFO_HOB, and graphics hardware information
+through EFI_PEI_GRAPHICS_DEVICE_INFO_HOB.
+
+EFI_PEI_GRAPHICS_INFO_HOB and EFI_PEI_GRAPHICS_DEVICE_INFO_HOB provide the basic information
+for the graphics display. These HOBs are described in the *PI Specification.*
+
+Please refer Appendix 6.6 EFI_PEI_GRAPHICS_INFO_HOB and 6.7 EFI_PEI_GRAPHICS_DEVICE_INFO_HOB for the details.
 
-ACPI Table HOB
+New Interfaces
 ~~~~~~~~~~~~~~
 
-The bootloader should pass ACPI table through the GUID HOB to the payload. So that the payload could get the platform information from the ACPI table.
+Common Payload Header
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All new interfaces are GUID type HOBs starting with ``EFI_HOB_GUID_TYPE`` defined in the PI Specification.
+
+The HOB data starts with a common header defined as below::
+
+  #pragma pack(1)
+  
+  typedef struct {
+    UINT16               Revision;

```

Comment:
Do we need 16 bit revision?   8 bit should be good enough ?

Reply:
> Do we need 16 bit revision? 8 bit should be good enough ?
0x0104 stands for 1.4?
I am fine to use one byte:)
then do you reserve

Reply:
The nest field still needs to be 4 byte aligned at least.

Reply:
sure. I will reserve 3 bytes.